pkg/issuer/vault/setup.go

```diff
@@ -69,21 +71,34 @@ func (v *Vault) Setup(ctx context.Context) error {
 	// check if only token auth method is set.
 	if v.issuer.GetSpec().Vault.Auth.TokenSecretRef.Name != "" &&
 		(v.issuer.GetSpec().Vault.Auth.AppRole.RoleId != "" ||
-			v.issuer.GetSpec().Vault.Auth.AppRole.SecretRef.Name != "") {
+			v.issuer.GetSpec().Vault.Auth.AppRole.SecretRef.Name != "" ||
+			v.issuer.GetSpec().Vault.Auth.Kubernetes.Role != "") {
 		klog.Infof("%s: %s", v.issuer.GetObjectMeta().Name, messageAuthFieldRequired)
 		apiutil.SetIssuerCondition(v.issuer, v1alpha1.IssuerConditionReady, v1alpha1.ConditionFalse, errorVault, messageAuthFieldRequired)
 		return nil
 	}
 
 	// check if all mandatory Vault appRole fields are set.
 	if v.issuer.GetSpec().Vault.Auth.TokenSecretRef.Name == "" &&
+		v.issuer.GetSpec().Vault.Auth.Kubernetes.Role == "" &&
 		(v.issuer.GetSpec().Vault.Auth.AppRole.RoleId == "" ||
 			v.issuer.GetSpec().Vault.Auth.AppRole.SecretRef.Name == "") {
 		klog.Infof("%s: %s", v.issuer.GetObjectMeta().Name, messageAuthFieldRequired)
 		apiutil.SetIssuerCondition(v.issuer, v1alpha1.IssuerConditionReady, v1alpha1.ConditionFalse, errorVault, messageAuthFieldRequired)
 		return nil
 	}
 
+	// check if all mandatory Kubernetes Auth fields are set.
+	if v.issuer.GetSpec().Vault.Auth.Kubernetes.Role != "" &&
+		v.issuer.GetSpec().Vault.Auth.Kubernetes.SecretRef.Name != "" &&
+		(v.issuer.GetSpec().Vault.Auth.AppRole.RoleId != "" ||
+			v.issuer.GetSpec().Vault.Auth.AppRole.SecretRef.Name != "" ||
+			v.issuer.GetSpec().Vault.Auth.TokenSecretRef.Name != "") {
```

Comment:
I'm not even going to try and parse this combination of `&&`, `(`, `)` and `||` to understand this 🙈